src/ClassComponent.js

```diff
@@ -26,6 +26,18 @@ export default class ClassComponent extends React.Component {
     }, 2000);
   }
 
+  setIntervalAgain() {
```

Comment:
Since we are writing this functionality twice I would rename this to setInterval and use it both in componendDidMount as well as handleNameUpdate. You will probably also have to bind this in the constructor like we did with the other method.